--- a/linux/drivers/media/video/tlg2300/pd-main.c	2011-02-27 11:25:48.000000000 +0000
+++ b/linux/drivers/media/video/tlg2300/pd-main.c	2011-02-26 06:44:12.000000000 +0000
@@ -36,7 +36,6 @@
 #include <linux/string.h>
 #include <linux/types.h>
 #include <linux/firmware.h>
-#include <linux/smp_lock.h>
 
 #include "vendorcmds.h"
 #include "pd-common.h"
@@ -453,14 +452,10 @@
 
 	device_init_wakeup(&udev->dev, 1);
 #ifdef CONFIG_PM
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 34)
-	pd->udev->autosuspend_disabled = 0;
-#endif
-	pd->udev->autosuspend_delay = HZ * PM_SUSPEND_DELAY;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 35)
+	pm_runtime_set_autosuspend_delay(&pd->udev->dev,
+			1000 * PM_SUSPEND_DELAY);
 	usb_enable_autosuspend(pd->udev);
 
-#endif
 	if (in_hibernation(pd)) {
 		INIT_WORK(&pd->pm_work, hibernation_resume);
 		schedule_work(&pd->pm_work);
@@ -490,15 +485,11 @@
 	/*unregister v4l2 device */
 	v4l2_device_unregister(&pd->v4l2_dev);
 
-	lock_kernel();
-	{
-		pd_dvb_usb_device_exit(pd);
-		poseidon_fm_exit(pd);
+	pd_dvb_usb_device_exit(pd);
+	poseidon_fm_exit(pd);
 
-		poseidon_audio_free(pd);
-		pd_video_exit(pd);
-	}
-	unlock_kernel();
+	poseidon_audio_free(pd);
+	pd_video_exit(pd);
 
 	usb_set_intfdata(interface, NULL);
 	kref_put(&pd->kref, poseidon_delete);
